plugins/onepad/Linux/dialog.cpp

```diff
@@ -698,5 +701,6 @@ void DisplayDialog()
 {
     Dialog* dialog = new Dialog();
     dialog->InitDialog();
-    dialog->Show(true);
+    dialog->ShowModal();
+    dialog->Destroy();
```

Comment:
Since `dialog` is now modal, it can be allocated on the stack instead. `Destroy()` won't need to be called since that'll automatically happen when it goes out of scope.